control-plane/api/v1alpha1/registration_webhook.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package v1alpha1
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/hashicorp/consul-k8s/control-plane/api/common"
+)
+
+// +kubebuilder:object:generate=false
+
+type RegistrationWebhook struct {
+	Logger logr.Logger
+
+	// ConsulMeta contains metadata specific to the Consul installation.
+	ConsulMeta common.ConsulMeta
+
+	decoder *admission.Decoder
+	client.Client
+}
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-v1alpha1-registration,mutating=false,failurePolicy=fail,groups=consul.hashicorp.com,resources=registrations,versions=v1alpha1,name=validate-registration.consul.hashicorp.com,sideEffects=None,admissionReviewVersions=v1beta1;v1
+
+func (v *RegistrationWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
+	resource := &Registration{}
+	decodeErr := v.decoder.Decode(req, resource)
+	if decodeErr != nil {
+		return admission.Errored(http.StatusBadRequest, decodeErr)
+	}
+
+	v.Logger.Info("validating registration", "name", resource.Name, "timeout", resource.Spec.HealthCheck.Definition.TimeoutDuration)
```

Comment:
~ Seeing as this HealthCheck field `TimeoutDuration string` is optional (not sure if we have a default timeout value), should it be added to the logger?

Reply:
oh whoops, this was a debugging statement